develop/dev-guide-third-party-tools-compatibility.md

```diff
@@ -0,0 +1,158 @@
+
+### TiDB distinguishes between `TIMESTAMP` and `DATETIME` in error messages
+
+**Description**
+
+TiDB error messages distinguish between `TIMESTAMP` and `DATETIME`, while MySQL does not, and returns them all as `DATETIME`. That is, MySQL incorrectly converts `TIMESTAMP` type error messages to `DATETIME` type.
+
+**Way to avoid**
+
+Do not use the error messages for string matching, but use [Error Codes](/error-codes.md) for troubleshooting.
+
+### TiDB does not support the `CHECK TABLE` statement
+
+**Description**
+
+The `CHECK TABLE` statement is not supported in TiDB. To check the consistency of data and corresponding indexes, you can use the [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) statement in TiDB.
+
+**Way to avoid**
+
+Use [`ADMIN CHECK [TABLE|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) instead.
+
+## MySQL JDBC incompatibility
+
+Test Version: MySQL Connector/J `8.0.29`
+
+### Default collations are inconsistent
+
+**Description**
+
+The collations of MySQL Connector/J are stored in the client and are distinguished by the server version.
+
+Known character sets for which client-side and server-side collations do not match:
+
+| Character Sets | Client Default Collations | Server Default Collations |
+| - | - | - |
+| `ascii` | `ascii_general_ci` | `ascii_bin` |
+| `latin1` | `latin1_swedish_ci` | `latin1_bin` |
+| `utf8mb4` | `utf8mb4_0900_ai_ci` | `utf8mb4_bin` |
+
+**Way to avoid**
+
+Set the sorting rules manually and do not rely on the client default sorting rules (client default sorting rules are saved by the MySQL Connector/J configuration file).
+
+### The `NO_BACKSLASH_ESCAPES` parameter is ineffective
+
+**Description**
+
+The `NO_BACKSLASH_ESCAPES` parameter cannot be used so that `\` character is not escaped. Tracking the [issue](https://github.com/pingcap/tidb/issues/35302).
+
+**Way to avoid**
+
+Do not use `NO_BACKSLASH_ESCAPES` with `\`, but use normal `\\` for SQL writing.
+
+### The `INDEX_USED` parameters are not supported
+
+**Description**
+
+TiDB does not set the `SERVER_QUERY_NO_GOOD_INDEX_USED` and `SERVER_QUERY_NO_INDEX_USED` parameters in the protocol. This will cause the following parameters to be returned inconsistently with the actual:
+
+- `com.mysql.cj.protocol.ServerSession.noIndexUsed()`
+- `com.mysql.cj.protocol.ServerSession.noGoodIndexUsed()`
+
+**Way to avoid**
+
+Do not use the `noIndexUsed()` and `noGoodIndexUsed()` functions.
+
+### The `enablePacketDebug` parameter is not supported
+
+**Description**
+
+TiDB does not support the [enablePacketDebug](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-connp-props-debugging-profiling.html) parameter, which is a MySQL Connector/J parameter used for debugging that will retain the packet's ring buffer. This might cause the connection to **UNEXPECTED CLOSE** and **DO NOT** turn it on.
```

Comment:
```suggestion
TiDB does not support the [enablePacketDebug](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-connp-props-debugging-profiling.html) parameter. It is a MySQL Connector/J parameter used for debugging that will retain the packet's ring buffer. This might cause the connection to **UNEXPECTED CLOSE**. **DO NOT** turn it on.
```